File: src/main/java/com/beat/global/common/dto/SuccessResponse.java
package com.beat.global.common.dto;

import com.beat.global.common.exception.base.BaseSuccessCode;

public record SuccessResponse<T>(
	int status,
	String message,
	T data
) {
	public static <T> SuccessResponse<T> of(final BaseSuccessCode baseSuccessCode, final T data) {
		return new SuccessResponse<>(baseSuccessCode.getStatus(), baseSuccessCode.getMessage(), data);
	}

	public static <T> SuccessResponse<T> from(final BaseSuccessCode baseSuccessCode) {
		return new SuccessResponse<>(baseSuccessCode.getStatus(), baseSuccessCode.getMessage(), null);
	}
}

File: global-support/src/main/kotlin/com/beat/global/support/response/SuccessResponse.kt
package com.beat.global.support.response

data class SuccessResponse<T>(
    val status: Int,
    val message: String,
    val data: T?,
) {
    companion object {
        @JvmStatic
        fun <T> of(
            successCode: SuccessCode,
            data: T,
        ): SuccessResponse<T> =
            SuccessResponse(successCode.getStatus(), successCode.getMessage(), data)

        @JvmStatic
        fun <T> from(successCode: SuccessCode): SuccessResponse<T> =
            SuccessResponse(successCode.getStatus(), successCode.getMessage(), null)
    }
}
